### PR TITLE
Add support for alluxio::file:// protocol

### PR DESCRIPTION
**Pull request by LuQQiu**
Add support for adding alluxio to the beginning of the full ufs url.
This is helpful for users that only want to use cache for some workloads of the target ufs but not others
Following the caching convention of https://filesystem-spec.readthedocs.io/en/latest/features.html#url-chaining

### CODE REVIEW

File: alluxiofs/core.py [hunk 1]
```diff
@@ -224,123 +245,110 @@ def cat_file(self, path, start=None, end=None, **kwargs):
...
+                    return fs_method(
+                        self._strip_alluxio_protocol(path), *args, **kwargs
+                    )
+                raise NotImplementedError(
+                    f"The method {func.__name__} is not implemented in the underlying filesystem."
+                )
+            else:
+                raise NotImplementedError(
+                    "Underlying filesystem is not available."
+                )
```
Comment (lucyge2022) [inline alluxiofs/core.py:263]: why dont we merge this with alluxio_with_fallback_handler?
coz those not implemented in fsspec is not implemented in alluxio is it? if alluxio throws unimplemented error then we check self.fs (fsspec ufs) and then if it still isnt present it throws notimplmented right?

File: alluxiofs/core.py [hunk 2]
```diff
@@ -120,7 +120,23 @@ def __init__(
             if preload_path is not None:
                 self.alluxio.load(preload_path)
 
+        def _strip_alluxio_protocol(path):
```
Comment (lucyge2022) [inline alluxiofs/core.py:123]: can u add doc here?
so this func is to turn
1) alluxio::s3://path  -> into  s3://path
2) s3://path -> into s3://path
and the _strip_protocol is to turn
1) alluxio::s3://path -> path
2) s3://path -> into path
am i correct? 
can u add some doc showing these examples?

**Review by lucyge2022 — APPROVED** [review]
LGTM

File: alluxiofs/core.py (hunk 2 above)
Comment (LuQQiu) [reply alluxiofs/core.py:123]: it only removes alluxio:: from any path

**Pull request merged**